File: billing-application/src/main/java/com/example/billing/application/command/IngestUsageCommand.java
package com.example.billing.application.command;

import com.example.billing.domain.metering.ResourceType;
import com.example.billing.domain.shared.CustomerId;

import java.time.Instant;
import java.util.UUID;

public record IngestUsageCommand(
        UUID eventId,
        CustomerId customerId,
        ResourceType resourceType,
        long quantity,
        Instant occurredAt
) {}

File: billing-application/src/main/kotlin/com/example/billing/application/command/IngestUsageCommand.kt
package com.example.billing.application.command

import com.example.billing.domain.metering.ResourceType
import com.example.billing.domain.shared.CustomerId
import java.time.Instant
import java.util.UUID

@JvmRecord
data class IngestUsageCommand(
    val eventId: UUID,
    val customerId: CustomerId,
    val resourceType: ResourceType,
    val quantity: Long,
    val occurredAt: Instant,
)
